gridt/tests/models/test_creation.py

```diff
@@ -0,0 +1,52 @@
+from gridt.tests.basetest import BaseTest
+from gridt.models import Creation, User, Movement
+from freezegun import freeze_time
+from datetime import datetime
+
+
+class UnitTestCreation(BaseTest):
+    def test_init(self):
+        creation1 = None
+        with freeze_time('2022-12-27 03:36:00'):
+            creation1 = Creation()
+
+        self.assertIsNone(creation1.movement)
+        self.assertIsNone(creation1.user)
+        self.assertEqual(creation1.time_added, datetime(2022,12,27,3,36,00))
+        self.assertEqual(creation1.type, 'creation')
+
+        user = self.create_user()
+        movement = self.create_movement()
+
+        creation2 = None
+        with freeze_time('2022-12-27 03:36:00'):
+            creation2 = Creation(user, movement)
+        
+        self.assertEqual(creation2.movement, movement)
+        self.assertEqual(creation2.user, user)
+        self.assertEqual(creation2.time_added, datetime(2022,12,27,3,36,00))
+        self.assertEqual(creation2.type, 'creation')
+
+
+    def test_end(self):
+        user = self.create_user()
+        movement = self.create_movement()
+        creation = Creation(user, movement)
+
+        with freeze_time('2022-12-27 03:36:00'):
+            creation.end()
+
+        self.assertEqual(creation.time_removed, datetime(2022,12,27,3,36,00))
+        self.assertTrue(creation.is_ended())
+
+
+    def test_repr(self):
+        user = User("abc123", "test@test.com", "password")
```

Comment:
I'm not sure about python but in java i like to put these hardcoded string as constants at the top of the class, because then you can reuse them.

Reply:
Hmm... ya makes sense

Reply:
Something I would do differently now looking back, is replacing the `unittest` library from python with [pytest](https://docs.pytest.org/en/7.2.x/), which is kind of the industry standard. In that case, you could create [fixtures](https://docs.pytest.org/en/7.2.x/explanation/fixtures.html#about-fixtures) to create the User. Having the strings inline there is not a problem, since you already abstracted the "User" entity creation to a different test. An equivalent way of dealing with this in unittest I guess, would be to have a separate non test method on the class where you would create the test user.

Reply:
I would say that this also forces you to test in a way that is more sensible/readable.